documentation/tests/integration/components/list-item-test.js

```diff
@@ -0,0 +1,151 @@
+  });
+
+  test('it should render a <a> link withhout "target" and "rel" attributes if @isHrefExternal is false', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" @isHrefExternal={{false}} />`
+    );
+    assert.dom('#test-list-item').doesNotHaveAttribute('target');
+    assert.dom('#test-list-item').doesNotHaveAttribute('rel');
+  });
+
+  test('it should spread all the attributes passed to the <div> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('div#test-list-item').hasClass('my-class');
+    assert.dom('div#test-list-item').hasAttribute('data-test1');
+    assert.dom('div#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <a> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <LinkTo> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem @route="index" id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <button> element', async function (assert) {
+    this.set('onClick', () => {});
+
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @onClick={{this.onClick}} class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('button#test-list-item').hasClass('my-class');
+    assert.dom('button#test-list-item').hasAttribute('data-test1');
+    assert.dom('button#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should yield the content of the List item element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" as |L|><L.Content>test</L.Content></Cut::ListItem>`
+    );
+    assert.dom('div#test-list-item > div.cut-list-item__content').exists();
+    assert
+      .dom('div#test-list-item > div.cut-list-item__content')
+      .hasText('test');
+  });
+
+  test('it should yield the action(button) of the List item element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" as |L|><L.Action as |A|><A.Button @text="test"/></L.Action></Cut::ListItem>`
+    );
+    assert.dom('div#test-list-item > div > button.hds-button').exists();
```

Comment:
I see a risk in relying on the DOM structure for the selector here: maybe you can assign an ID to the `<A.Button>` and do something like this:
```suggestion
    assert.dom('#test-list-item #ID-OF-THE-BUTTON').exists();
```